run-tests.php

```diff
@@ -2160,22 +2160,42 @@ function run_test($php, $file, $env)
 				return 'SKIPPED';
 			}
 
+			$keywordMissing = true;
+
 			if (!strncasecmp('info', ltrim($output), 4)) {
 				if (preg_match('/^\s*info\s*(.+)\s*/i', $output, $m)) {
 					$info = " (info: $m[1])";
+					$keywordMissing = false;
 				}
 			}
 
 			if (!strncasecmp('warn', ltrim($output), 4)) {
 				if (preg_match('/^\s*warn\s*(.+)\s*/i', $output, $m)) {
 					$warn = true; /* only if there is a reason */
 					$info = " (warn: $m[1])";
+					$keywordMissing = false;
 				}
 			}
 
 			if (!strncasecmp('xfail', ltrim($output), 5)) {
 				// Pretend we have an XFAIL section
 				$section_text['XFAIL'] = trim(substr(ltrim($output), 5));
+				$keywordMissing = false;
+			}
+
+			if ($keywordMissing && trim($output) !== '') {
```

Comment:
Might be more elegant to convert this into an elseif chain?

Reply:
done